UIE-204 Narrow Ajax Usage pt3

msilva-broad opened the pull request:
- switch first wave of shared src/workspaces code to call Ajax sub-modules directly.
- updated related unit test mocks.
- updated workflows WorkflowView tests to honor Ajax shifts in useWorkspace related changes, and shifted its own Ajax calls.

### Jira Ticket: https://broadworkbench.atlassian.net/browse/[Ticket #]

<!-- ### Dependencies --->
<!-- Include any dependent tickets and describe the relationship. Include any relevant Jira tickets. --->

## Summary of changes:
<!--Please give an abridged version of the ticket description here and/or fill out the following fields.-->

### What
-

### Why
- improve code maintainability and potential for code sharing.

### Testing strategy
<!-- Note that changes impacting components in Storybook stories can be viewed at
https://www.chromatic.com/library?appId=65fc89c9335768720ff8605a&branch=<branch>
The branch corresponding to this PR is selected, and changes can be reviewed by commit. --->

- [ ] <!-- Test case 1 -->

<!-- ### Visual Aids -->
<!-- https://support.apple.com/guide/quicktime-player/record-your-screen-qtp97b08e666/mac -->

msilva-broad committed:
UIE-204 Narrow Ajax Usage pt3
- switch first wave of shared src/workspaces code to call Ajax sub-modules directly.
- updated related unit test mocks.
- updated workflows WorkflowView tests to honor Ajax shifts in useWorkspace related changes, and shifted its own Ajax calls.

msilva-broad committed:
UIE-204 Narrow Ajax Usage pt3
- updated WorkspaceTags ajax usage.

cahrens commented on src/pages/workspaces/workspace/workflows/WorkflowView.js:
```diff
@@ -507,7 +511,7 @@ export const WorkflowView = _.flow(
                   // will only match if the current root entity type comes from a snapshot
                   const snapshot = _.find({ metadata: { name: modifiedConfig.dataReferenceName } }, availableSnapshots);
-                  Ajax().Metrics.captureEvent(Events.workflowLaunch, {
+                  void Metrics().captureEvent(Events.workflowLaunch, {
```
Does the void make it clear that we don't care about awaiting?

msilva-broad replied:
correct, and it removes a lint warning for the same reason.

I may have missed a couple of these in a recent PR, but try to fix them as I see them.

cahrens replied:
Cool! It would be good to do it consistently then.

msilva-broad replied:
ok yeah, I see a few I missed in other PR files.... I'll fix um now

msilva-broad committed:
UIE-204 Narrow Ajax Usage pt3
- added missing void keyword to other fire-and-forget async calls (mostly Metrics calls)

sam-schu reviewed (approved):
Workflows changes look good!
